idna/src/uts46.rs

```diff
@@ -223,6 +225,36 @@ fn passes_bidi(label: &str, is_bidi_domain: bool) -> bool {
             }
         }
 
+        BidiClass::EN => {
+            // https://github.com/servo/rust-url/issues/489
+            // LDH labels that start with a digit are allowed when they don't come after RTL label
+            if *after_rtl_label  {
+                return false;
+            }
+            match first_char {
+                Some(c) if c.is_ascii() => {},
+                _ => { return false; }
+            };
+            // check that label is LDH
+            // https://tools.ietf.org/html/rfc5890#section-2.3.1
+            let mut last = chars.next();
+            loop {
+                match last {
+                    Some(c) => {
+                        if !c.is_ascii() {
+                            return false;
+                        }
+                        last = chars.next();
+                    }
+                    _ => { break; }
+                }
+            }
```

Comment:
This loop doesn't really check if it is truly a LDH label. I think it would accept something like `3_4` as well (`_` is ascii, but neither letter, digit or hyphen). Does something else check that in the existing code?